### DISCUSSION TITLE
Radiolib with custom board using STM32L0 and SX1262

### DISCUSSION THREAD

**Discussion by nikorainto** [jgromes/RadioLib#1127] [Q&A] [answered]
Hello!

First of all big thanks for making this library. It seems really robust, supports many devices and is also well maintained

I have a problem with my custom board. I'm trying to use general [lora example](https://github.com/jgromes/RadioLib/blob/master/examples/SX126x/SX126x_Transmit_Blocking/SX126x_Transmit_Blocking.ino) to test that my board is able to send lora messages. I'm able to get past `radio.begin()` but when trying to send test message with `radio.transmit()` it always ends in `RADIOLIB_ERR_TX_TIMEOUT`. I start to believe I have something wrong with my interrupt/GPIO configuration

Here is how my schematics look for [STM32L072RBT6](https://www.st.com/en/microcontrollers-microprocessors/stm32l072rb.html) and [Waveshare SX1262](https://www.waveshare.com/core1262-868m.htm) connection:

![image](https://github.com/jgromes/RadioLib/assets/26547857/c40c52b3-e43a-403e-b96a-424f56d6da13)

And here is repo for my current code: https://github.com/nikorainto/savi-lorawan-radiolib

Would someone be able to help me detect what am I missing or doing wrong? I would really appreciate it and like to use this library to test my custom board 😄 

Note: My board is really bare-bone and it does not have serial port or debug leds 

I also made a [post](https://community.platformio.org/t/lorawan-library-for-sx1262/40469) about this on PlatformIO forums and there I got help to progress but I'm still stuck with this `RADIOLIB_ERR_TX_TIMEOUT` issue

**Comment by jgromes** [comment]
The Tx timeout error is caused when the radio does not trigger the DIO1 within the expected time frame, usually because DIO1 is not connected. Since the schematic you posted claims that "DIO1 and DIO2 are hooked the wrong way round", the first thing I would try is swapping DIO1 and DIO2. Try both options and see which one works - experience says that notes on schematics about swapped are correct about 50% of the time.

**Reply by nikorainto** [reply]
Sorry forgot to mention that I have already tried using PA2 instead of PA3 on `SX1262 radio = new Module(PA4, PA3, PC4, PC5);` multiple times but same problem. Also I soldered another board just to verify that there is no cold solder joint on DIO1 & DIO2

**Reply by jgromes** [reply]
In that case you could use an oscilloscope to verify whether DIO is actually being triggered. Another thing to consider is the time measurement - maybe your board has incorrectly configured clocks, so it doesn't measure time correctly?

**Reply by nikorainto** [reply]
Thanks for the tips! I will first try to verify clock configuration since I don't have oscilloscope available

**Reply by nikorainto** [reply]
Now I think I found something that could explain this. I started to go through design file and saw this:

![image](https://github.com/jgromes/RadioLib/assets/26547857/1e34516d-a12f-4c7f-899d-2c1c38f223e4)

In schematic above RXEN = 14, TXEN = 15, DIO1 = 16, DIO2 = 17 and I have gone all the time with pin map from Schematics and not even paid attention to the design 😬

**Reply by nikorainto** [reply, marked as answer]
Yes verified that I was now able to run lora example with following pin configuration

```
SX1262 radio = new Module(PA4, PA0, PC4, PC5);

radio.setRfSwitchPins(PA3, PA2);
```
